### PR TITLE
Issue 34/categories front

### LINKED ISSUE #34: Category creating and index

### PR DESCRIPTION
**Pull request by AlexDashko1337**
# Description

Added creating and index category pages 

Fixes # (issue)
https://github.com/nromanen/Ch-133.Ruby/issues/34

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,227 @@
+
```
Comment (bearded) [inline .gitignore:1]: it seems like this file should not be added

File: .idea/dataSources.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bearded) [inline .idea/dataSources.xml]: it seems like this file should not be added

File: server/config/database.yml [hunk 3]
```diff
@@ -17,8 +17,8 @@
 default: &default
   adapter: postgresql
   encoding: unicode
-  username: <%= ENV["DB_USER"] %>
-  password: <%= ENV["DB_PASSWORD"] %>
+  username: postgres
+  password: toor
   host: localhost
```
Comment (bearded) [inline server/config/database.yml]: those changes should not be committed

File: server/config/environments/development.rb [hunk 4]
```diff
@@ -43,8 +43,8 @@
   config.action_mailer.perform_deliveries = true
   config.action_mailer.delivery_method = :smtp
   config.action_mailer.smtp_settings = {
-    user_name: "#{ENV["SMTP_EMAIL"]}",
-    password: "#{ENV["SMTP_EMAIL_PASSWORD"]}",
+    user_name: "advertisements.smtp@gmail.com",
+    password: "Advert-smtp",
     domain: "localhost:3000",
```
Comment (bearded) [inline server/config/environments/development.rb]: those changes should not be committed

File: server/config/routes.rb [hunk 5]
```diff
@@ -8,6 +8,7 @@
                confirmations: "users/confirmations",
                unlocks: "users/unlocks"
              }
+<<<<<<<<< Temporary merge branch 1
```
Comment (bearded) [inline server/config/routes.rb]: needless line after conflict resolution

**Review by le-murier — APPROVED** [review]

**Pull request merged**